Assets/MixedRealityToolkit-SDK/Features/UX/Interactable/Scripts/Layout/ButtonLayout.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX
+{
+    /// <summary>
+    /// Place an object in space relative to another object's scale
+    /// Good for responsive buttons that can stretch and object realign
+    /// </summary>
+    [ExecuteInEditMode]
+    public class ButtonLayout : MonoBehaviour
+    {
+        [Tooltip("Where to set this object's center point in relation to the Anchor's center point")]
+        public Vector3 Alignment;
+
+        [Tooltip("A pixel to Unity unit conversion, Default: 2048x2048 pixels covers a 1x1 Unity Unit or default primitive size")]
+        public float BasePixelSize = 2048;
```

Comment:
Feels weird that this appears in several places. Should it be in a single place the project profiles? Would someone really want to set it to one value on a ButtonLayout and another on a ButtonCollider?

Reply:
It's more than just code, it's a workflow. I like the camera settings idea, but it seems too detacted from Interactables. Maybe changing the name of these files would reposition as more than just button related code, which could make more sense in a camera resolution type setting. This layout code is more about 3D layout in general then just button code anyway, with similar functionality to RectTransforms, but in 3D. I just crammed it in here because I use it for buttons and this was the first enty point. It should be rethought. We could call it Layout (ObjectSize, ObjectAlignment, ObjectSizeOffset, ColliderCopy, ObjectBorder)